rcl_yaml_param_parser/src/parser.c

```diff
@@ -0,0 +1,1384 @@
+        }
+      }
+      break;
+    case DATA_TYPE_INT64:
+      if (false == is_seq) {
+        param_value->integer_value = (int64_t *)ret_val;
+      } else {
+        if (DATA_TYPE_UNKNOWN == *seq_data_type) {
+          *seq_data_type = val_type;
+          param_value->integer_array_value =
+            allocator.zero_allocate(1U, sizeof(rcl_int64_array_t), allocator.state);
+          if (NULL == param_value->integer_array_value) {
+            RCUTILS_SAFE_FWRITE_TO_STDERR("Error allocating mem");
+            return RCL_RET_BAD_ALLOC;
+          }
+        } else {
+          if (*seq_data_type != val_type) {
+            RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(allocator, "Sequence should be of same"
+              " type. Value type 'integer' do not belong at line_num %d", line_num);
+            allocator.deallocate(ret_val, allocator.state);
+            return RCL_RET_ERROR;
+          }
+        }
+        res = add_val_to_int_arr(param_value->integer_array_value, ret_val, allocator);
+        if (RCL_RET_OK != res) {
+          if (NULL != ret_val) {
+            allocator.deallocate(ret_val, allocator.state);
+          }
+          return res;
+        }
+      }
+      break;
+    case DATA_TYPE_DOUBLE:
+      if (false == is_seq) {
+        param_value->double_value = (double *)ret_val;
+      } else {
+        if (DATA_TYPE_UNKNOWN == *seq_data_type) {
+          *seq_data_type = val_type;
+          param_value->double_array_value =
+            allocator.zero_allocate(1U, sizeof(rcl_double_array_t), allocator.state);
+          if (NULL == param_value->double_array_value) {
+            RCUTILS_SAFE_FWRITE_TO_STDERR("Error allocating mem");
+            return RCL_RET_BAD_ALLOC;
+          }
+        } else {
+          if (*seq_data_type != val_type) {
+            RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(allocator, "Sequence should be of same"
+              " type. Value type 'double' do not belong at line_num %d", line_num);
+            allocator.deallocate(ret_val, allocator.state);
+            return RCL_RET_ERROR;
+          }
+        }
+        res = add_val_to_double_arr(param_value->double_array_value, ret_val, allocator);
+        if (RCL_RET_OK != res) {
+          if (NULL != ret_val) {
+            allocator.deallocate(ret_val, allocator.state);
+          }
+          return res;
+        }
+      }
+      break;
+    case DATA_TYPE_STRING:
+      if (false == is_seq) {
+        param_value->string_value = (char *)ret_val;
+      } else {
+        if (DATA_TYPE_UNKNOWN == *seq_data_type) {
+          *seq_data_type = val_type;
+          param_value->string_array_value =
+            allocator.zero_allocate(1U, sizeof(rcutils_string_array_t), allocator.state);
+          if (NULL == param_value->string_array_value) {
+            RCUTILS_SAFE_FWRITE_TO_STDERR("Error allocating mem");
+            return RCL_RET_BAD_ALLOC;
+          }
+        } else {
+          if (*seq_data_type != val_type) {
```

Comment:
This line resulted in a new compiler warning in [this job](https://ci.ros2.org/view/packaging/job/packaging_linux-aarch64/451/warnings23Result/new/). Please address the new warning as soon as feasible. Thanks.

Reply:
Looking at it now

Reply:
#245